### PR TITLE
fix shutdown for PeriodicLog

### PR DESCRIPTION
**Pull request by CheSema**
It just not right when `PeriodicLog` inherits `SystemLog` and overrides `shutdown` method but do not fully shutdown the `SystemLog`.

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit a9e793532ae308767da3bc4da74d9631cd85eb90 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/68728/a9e793532ae308767da3bc4da74d9631cd85eb90/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/00aa60ca032f2b858579bd8c220bfcebfbe57f09/docs_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: src/Interpreters/PeriodicLog.cpp [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #include <Interpreters/PeriodicLog.h>
 #include <Interpreters/ErrorLog.h>
 #include <Interpreters/MetricLog.h>
+#include "Functions/DateTimeTransforms.h"
```
Comment (pamarcos) [inline src/Interpreters/PeriodicLog.cpp:4]: This include is not needed. By the looks of it, seems it was automatically added by VSCode or some other LSP?

File: src/Common/SystemLogBase.cpp [hunk 2]
```diff
@@ -273,6 +273,25 @@ void SystemLogBase<LogElement>::startup()
     saving_thread = std::make_unique<ThreadFromGlobalPool>([this] { savingThreadFunction(); });
 }
 
+template <typename LogElement>
+void SystemLogBase<LogElement>::stopFlushThread()
```
Comment (CheSema) [inline src/Common/SystemLogBase.cpp:277]: this change is minor.

File: src/Interpreters/PeriodicLog.cpp [hunk 3]
```diff
@@ -19,15 +20,15 @@ void PeriodicLog<LogElement>::stopCollect()
...
+    if (collecting_thread)
+        collecting_thread->join();
 }
 
 template <typename LogElement>
 void PeriodicLog<LogElement>::shutdown()
 {
     stopCollect();
-    this->stopFlushThread();
+    Base::shutdown();
```
Comment (CheSema) [inline src/Interpreters/PeriodicLog.cpp:31]: this is major change.

**Review by pamarcos — COMMENTED** [review]
Good catch! Should have felt confident enough when I refactored `MetricLog` (prior code [here](https://github.com/ClickHouse/ClickHouse/blob/c35a4364359e0d0d2b061c684b3005c5e179b5e5/src/Interpreters/MetricLog.cpp#L78-L82)) to do exactly the same change, because I remember thinking why would the shutdown of `MetricLog` avoid the `table->flushAndShutdown()`.

After your change, it's calling `StorageMergeTree::shutdown` as it should have always called :+1:

File: src/Interpreters/PeriodicLog.cpp (hunk 1 above)
Comment (CheSema) [reply src/Interpreters/PeriodicLog.cpp:4]: Yes, is is automatically added. I missed that, i will remove it, thanks.

**Review by pamarcos — APPROVED** [review]

**Comment by pamarcos** [comment]
Funny enough, I found I had this issue fixed as part of other PR in which I modified a little bit the `PeriodicSystemLog` as well :man_facepalming: : https://github.com/ClickHouse/ClickHouse/pull/66532/files#diff-244a9280796546df36294ffbb489ce816cb3eadb49b9efda9b9fc5073127a312R36

**Pull request merged**